lib/facility.rb

```diff
@@ -1,14 +1,89 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+               :r_vehicle,
+               :total
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @r_vehicles = []
+    @total = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def registered_vehicles 
+    @r_vehicles
+  end 
+
+  def collected_fees
+    @total
+  end
```

Comment:
If you created `attr_readers` for these attributes, you would not need to create these methods. You would, however, need to change the name of your instance variables in your initialize method to be named the way you want the method to be named.